controllers/progressiverollout_controller_test.go

```diff
@@ -0,0 +1,93 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	deploymentskyscannernetv1alpha1 "github.com/Skyscanner/argocd-progressive-rollout/api/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"time"
+)
+
+const (
+	timeout  = time.Second * 10
+	interval = time.Millisecond * 100
+)
+
+var _ = Describe("ProgressiveRollout Controller", func() {
+
+	apiGroup := "argoproj.io/v1alpha1"
+	ctx := context.Background()
+	// See https://onsi.github.io/gomega#modifying-default-intervals
+	SetDefaultEventuallyTimeout(timeout)
+	SetDefaultEventuallyPollingInterval(interval)
+
+	BeforeEach(func() {
+		namespace := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{Name: "argocd"},
+		}
+		err := k8sClient.Create(context.Background(), &namespace)
+		Expect(err).NotTo(HaveOccurred(), "failed to create test namespace")
+	})
+
+	// We should replace this test with better ones as we deploy the controller
+	It("should reconcile", func() {
+		pr := deploymentskyscannernetv1alpha1.ProgressiveRollout{
+			ObjectMeta: metav1.ObjectMeta{Name: "go-infrabin", Namespace: "argocd"},
```

Comment:
Does the reference to go-infrabin have to be here?

Reply:
No, I just had to invent a name, so maybe having a standard test name?

Reply:
IMO it would be better to define a standard test name and stick to it, reference to something existing may create confusion.